### DISCUSSION TITLE
The temperature only compute once in thermo-mechanical analysis

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#18326] [Q&A General] [answered]
Hi there,

I'm trying on a simple thermo-mechanical analysis of a rectangle plate, with the initial temperature 600K and the boundaries temperature as 300K, but the results shows the temperature only compute one time step and don't change anymore. What's missing of the problem? The input file is as follows. Thank you.
```
_[GlobalParams]
  displacements = 'disp_x disp_y'
[]
[Mesh]
  type = FileMesh
  file = 'plate/plate.msh'
[]
[Variables]
  [disp_x]
  []
  [disp_y]
  []
  [temperature]
    initial_condition = 600.0
  []
[]
[Modules/TensorMechanics/Master]
  [all]
    add_variables = true
    strain = SMALL
    eigenstrain_names = eigenstrain
    use_automatic_differentiation = true
    generate_output = 'vonmises_stress elastic_strain_xx elastic_strain_yy strain_xx strain_yy'
  []
[]
[Kernels]
  [heat_conduction]
    type = ADHeatConduction
    variable = temperature
    thermal_conductivity = k
  []
  [heat_conduction_time_derivative]
    type = ADHeatConductionTimeDerivative
    variable = temperature
    density_name = rho
    specific_heat = c
  []
[]
[BCs]
  [x_symmetry]
    type = DirichletBC
    variable = disp_x
    boundary = right
    value = 0
  []
  [heatflux_symmetry]
    type = NeumannBC
    variable = temperature
    boundary = right
    value = 0
  []
  [Temperature_fix_left]
    type = DirichletBC
    variable = temperature
    boundary = left
    value = 300
  []
  [Temperature_fix_top]
    type = DirichletBC
    variable = temperature
    boundary = top
    value = 300
  []
  [Temperature_fix_bottom]
    type = DirichletBC
    variable = temperature
    boundary = bottom
    value = 300
  []
[]

[Materials]
  [bulk]
    type = ADGenericConstantMaterial
    prop_names = 'rho c k'
    prop_values = ' ${rho} ${c} ${k}'
  []
  [./elasticity_tensor]
    type = ADComputeIsotropicElasticityTensor
    youngs_modulus = 218.4e3
    poissons_ratio = 0.2
  [../]
  [./stress]
    type = ADComputeLinearElasticStress
  [../]
  [thermal_strain]
    type = ADComputeThermalExpansionEigenstrain
    stress_free_temperature = 300
    eigenstrain_name = eigenstrain
    temperature = temperature
    thermal_expansion_coeff = 7.5e-6
  []
[]
[Executioner]
  type = Transient
  solve_type = PJFNK
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
  petsc_options_value = 'lu       superlu_dist                 '
  automatic_scaling = true
  dt = 0.1e-6
  end_time = 200e-6
  nl_rel_tol = 1e-08
  nl_abs_tol = 1e-10

  picard_max_its = 100
  picard_abs_tol = 1e-50
  picard_rel_tol = 1e-03
  accept_on_max_picard_iteration = true
[]
[Outputs]
  exodus = true
[]__
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

you expect the solution to be slow cooling from 600 to 300K ?
What does it stop at?

btw 
- you have a bunch of deprecated Picard iteration parameters at the bottom of your input. Use `fixed_point_` instead of `picard_` there.
If you dont have a multiapps setup I dont think you need them either. 

- when using automatic_differentiation, you want to use Newton to solve the system, not PJFNK, because AD provides you with a good jacobian

We'll tag some mechanics people on this

Guillaume

**Reply by rh201** [reply]
Thank you.
Yes, I want to cool it down, and the simulation continues to the end, but the result shows the temperature only changes from 0 to the first time step and doesn't change in the following steps.
Yes, the code above is part of a multiapps setup, which has problems. I'm trying to figure what's wrong by commenting multiapps part out. Thanks for reminding that.

**Reply by GiudGiud** [reply]
The initial condition of the temperature is 600 not 0 though?

**Reply by rh201** [reply]
Yes, the initial conditions of the temperature is set as 600, cooling to 300

**Reply by GiudGiud** [reply]
oh ok I misread 0 as the initial value not the time.

@hugary1995 any idea what is going wrong here?

**Reply by ABallisat** [reply]
What are your thermal material properties and how long do you expect it to take to cool down? You are only running your model for 200E-6 seconds so for it to get to equilibrium 300 K in that time would require a mean cooling rate of 1.5 million K/s, is that what you are intending? Do you need to run at such a small time step? I suspect your time scales need to be closer to orders of 10s of seconds to see some temperature changes in its current form.

**Reply by rh201** [reply]
Thanks for replying. 
Yes, that's a good point, I'll try that. But even though the end time is small, it has 2000 time steps, the question is the temperature only changes in the first time step and keeps the same in the rest.

**Reply by GiudGiud** [reply]
Does changing the time step fix it?

If not, I dont see the definitions for ${rho} ${c} ${k}' in your input file? What are those values? Extreme values here could affect the solution

**Reply by ABallisat** [reply]
So the first time step is effectively finding the initial solution that meets your initial conditions (as my understanding works, correct me if I'm wrong). In the rest of your current scenario the temperature will be changing but very insignificantly - have you checked the numbers to see if they are changing at all? The number of time steps does not matter, it is the total time. Think of it like this: if you cook something in the oven at 200 C (so roughly 500 K), when you take it out and serve it at room temperature (roughly 300 K), it will stay warm for a long time (many minutes). How often you check it (equivalent to the number of time steps) does not matter as the physics doesn't care how often you measure it, it is how long you leave it for (the total model time) that matters.

**Reply by rh201** [reply]
@GiudGiud @ABallisat Yes, it is the matter of time scale. When I set dt=1e-3 and total time = 200e-3, the temperature transfers all the way to the end. But what I do not quite understand is when the total time is smaller, shouldn't the temperature change more dramatically than a longer time scale? Why does it not change anymore in the rest steps of the total time of 200e-6?

**Reply by GiudGiud** [reply]
changing how long you simulate the system does not change the way the real system would evolve. 
If it takes 10 second to cool down by 50K in the real world, then if you only simulate 200 micros you won't see much of that cooling happen.
